oop1-construct.js

```diff
@@ -20,58 +118,67 @@ Create a class that can categorize at least 6 of the creatures that you recall o
 
 class Shape {
     //your code here...
+    constructor(sides){
+        this.sides = sides; //lengths of each side
+    }
 }
 
 
-const triangle = {
-    name : "triangle",
-    sides : [4,7,7], //lengths of each side
-    base : null,
-    height : null, //determine this height using basic geometry.  You may need to research this one.
-    calcArea : function() {
-        console.log(`${this.name}'s area is calculated to be : ${.5 * this.base * this.height}`);
-    },
-    calcPerimeter : function() {
-        console.log(`${this.name}'s perimeter is calculated to be : ${(this.base) + (this.sides[1] + this.sides[2]) }`)
+class Triangle extends Shape {
+    base = this.sides[0];
+    height = null; //determine this height using basic geometry.  You may need to research this one.
+
+    calcArea() {
+        console.log(`${this.constructor.name}'s area is calculated to be : ${.5 * this.base * this.height}`);
+    }
+
+    calcPerimeter() {
+        console.log(`${this.constructor.name}'s perimeter is calculated to be : ${(this.base) + (this.sides[1] + this.sides[2]) }`)
     }
 }
 
+const triangle = new Triangle([4,7,7]);
+
 triangle.base = triangle.sides[0];  //base is the 1st side in the triangle.sides array.
-triangle.height = 6.71; //determine the height using basic geometry.  How do you calculate the height of a triangle with 2 equal sides?  If you use the formula to caluclate the height, this is a bonus.  If you hard code the correct value here, that will be sufficient, but no bonus :(
+triangle.height = Math.sqrt(Math.pow(triangle.sides[1], 2) - Math.pow((triangle.base / 2), 2)); //determine the height using basic geometry.  How do you calculate the height of a triangle with 2 equal sides?  If you use the formula to caluclate the height, this is a bonus.  If you hard code the correct value here, that will be sufficient, but no bonus :(
```

Comment:
This only works with an equilateral triangle. There's a formula called Heron's formula to calculate the height and area of a triangle with only the sides known.

MORE IMPORTANTLY, you want to put the formula for calculating height in the Triangle class. As it is now, you're just asking for a user to try to use a null value and crash your program.